crates/walrus-sdk/src/client/quilt_client.rs

```diff
@@ -0,0 +1,583 @@
+    encoding::{Primary, QuiltError, Secondary, SliverData, quilt_encoding::*},
+    metadata::{QuiltIndex, QuiltMetadata, QuiltMetadataV1, VerifiedBlobMetadataWithId},
+};
+use walrus_sui::client::{BlobPersistence, PostStoreAction, ReadClient, SuiContractClient};
+
+use crate::{
+    client::{Client, client_types::StoredQuiltPatch, responses::QuiltStoreResult},
+    error::{ClientError, ClientErrorKind, ClientResult},
+    store_when::StoreWhen,
+};
+
+// Reads all files recursively from a given path and returns them as path-content pairs.
+///
+/// If the path is a file, it's read directly.
+/// If the path is a directory, its files are read recursively.
+/// Returns error if path doesn't exist or is not accessible.
+pub fn read_blobs_from_paths<P: AsRef<Path>>(paths: &[P]) -> ClientResult<Vec<(PathBuf, Vec<u8>)>> {
+    if paths.is_empty() {
+        return Ok(Vec::new());
+    }
+
+    let mut collected_files: HashSet<PathBuf> = HashSet::new();
+    for path in paths {
+        let path = path.as_ref();
+
+        // Validate path existence and accessibility.
+        if !path.exists() {
+            return Err(ClientError::from(ClientErrorKind::Other(
+                format!("Path does not exist: {:?}.", path).into(),
+            )));
+        }
+
+        if path.is_file() {
+            collected_files.insert(path.to_path_buf());
+        } else if path.is_dir() {
+            collected_files.extend(get_all_files_from_dir(path)?);
+        } else {
+            return Err(ClientError::from(ClientErrorKind::Other(
+                format!("Path is neither a file nor a directory: {:?}.", path).into(),
+            )));
+        }
+    }
+
+    let mut collected_files_with_content: Vec<(PathBuf, Vec<u8>)> = Vec::new();
+    for file_path in collected_files {
+        let content = fs::read(&file_path).map_err(ClientError::other)?;
+        collected_files_with_content.push((file_path, content));
+    }
+
+    Ok(collected_files_with_content)
+}
+
+/// Get all file paths from a directory recursively.
+fn get_all_files_from_dir<P: AsRef<Path>>(path: P) -> ClientResult<Vec<PathBuf>> {
+    let path = path.as_ref();
+    let mut collected_files: HashSet<PathBuf> = HashSet::new();
+    let dir_entries = fs::read_dir(path).map_err(ClientError::other)?;
+    for entry in dir_entries {
+        let current_entry_path = entry.map_err(ClientError::other)?.path();
+        if current_entry_path.is_file() {
+            collected_files.insert(current_entry_path);
+        } else if current_entry_path.is_dir() {
+            collected_files.extend(get_all_files_from_dir(&current_entry_path)?);
+        }
+    }
+
+    Ok(collected_files.into_iter().collect())
+}
+
+/// A facade for interacting with Walrus quilt.
+#[derive(Debug, Clone)]
+pub struct QuiltClient<'a, T> {
+    client: &'a Client<T>,
+}
+
+impl<'a, T> QuiltClient<'a, T> {
+    const MAX_RETRIEVE_SLIVERS_ATTEMPTS: usize = 2;
+    const TIMEOUT_DURATION: Duration = Duration::from_secs(10);
```

Comment:
Suggestion: Make these configurable by changing them to struct fields that need to be provided when instantiating the `QuiltClient`. They can then later be provided through command-line options or the `client_config.yaml` with these as defaults.